pkg/provision/workspace/deployment.go

```diff
@@ -197,6 +197,7 @@ func getSpecDeployment(
 						constants.DevWorkspaceIDLabel:   workspace.Status.DevWorkspaceId,
 						constants.DevWorkspaceNameLabel: workspace.Name,
 					},
+					Annotations: workspace.Config.Workspace.PodAnnotations,
```

Comment:
I noticed we are [collecting the pod additions annotations](https://github.com/devfile/devworkspace-operator/blob/main/pkg/provision/workspace/deployment.go#L302-L304) but never actually using them. Currently, this is not a big deal, since annotations are never gathered while creating the pod additions list. I believe this logic was added to `mergePodAdditions` for completeness, but hasn't been used yet.

We could merge the `podAdditions.aAnnotations` with the `workspace.Config.Workspace.PodAnnotations`, or just use the workspace.Config.Workspace.PodAnnotations alone as there won't be any actual difference. Though in the future, this might have to make this change if the podAddtions bring in any pod annotations.

Reply:
Sounds good, I've updated to use `mergePodAdditions`